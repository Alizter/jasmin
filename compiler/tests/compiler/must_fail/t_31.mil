// ssa form, use before def

extern fn test(cf : reg bool) {
  x,y,z : reg u64;
  
  if cf {
    y = 0;
  } else {
    y = 1;
  }
  x = y; // this is OK
  x += z;
}

/*
START:CMD
ARG="renumber_fun_unique,typecheck,save[/tmp/ssa1.mil][]"
ARG="$ARG,merge_blocks,local_ssa[test]"
END:CMD
*/
