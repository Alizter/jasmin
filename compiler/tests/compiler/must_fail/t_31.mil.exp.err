Parsing tests/compiler/must_fail/t_31.mil as MIL

tests/compiler/must_fail/t_31.mil:12:8: 12:9 error: variable used before definition
  x += z;
       ^^
