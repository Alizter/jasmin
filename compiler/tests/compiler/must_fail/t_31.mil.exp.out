renumbering identifiers in all function(s)
   REMOVED_TIME
type checking module in all function(s)
   REMOVED_TIME
merging basic blocks in all function(s)
   REMOVED_TIME
transforming into local SSA form in test function(s)
